File: ios-native/KPKNFit/KPKNFit/Telemetry/TelemetryEvents.swift
enum TelemetryEvents {
    static let appOpen = "app_open"
    static let appForeground = "app_foreground"
    static let appBackground = "app_background"

    static let loginSuccess = "login_success"
    static let loginFailed = "login_failed"
    static let logout = "logout"

    static let workoutStart = "workout_start"
    static let workoutEnd = "workout_end"
    static let workoutPause = "workout_pause"
    static let workoutResume = "workout_resume"
    static let workoutCancel = "workout_cancel"
    static let setComplete = "set_complete"
    static let restStart = "rest_start"
    static let restEnd = "rest_end"
    static let exerciseComplete = "exercise_complete"

    static let nutritionOpen = "nutrition_open"
    static let mealLogStart = "meal_log_start"
    static let mealLogComplete = "meal_log_complete"
    static let foodSearch = "food_search"
    static let foodItemAdd = "food_item_add"
    static let nutritionParsePhoto = "nutrition_parse_photo"
    static let nutritionParseSuccess = "nutrition_parse_success"
    static let nutritionParseFailed = "nutrition_parse_failed"

    static let programSelect = "program_select"
    static let programStart = "program_start"
    static let programComplete = "program_complete"
    static let programCancel = "program_cancel"

    static let exerciceSearch = "exercise_search"
    static let exerciceViewDetails = "exercise_view_details"
    static let exerciceAddToWorkout = "exercise_add_to_workout"

    static let wikilabOpen = "wikilab_open"
    static let wikilabExerciseView = "wikilab_exercise_view"

    static let settingsOpen = "settings_open"
    static let settingsChange = "settings_change"
    static let themeChange = "theme_change"
    static let languageChange = "language_change"

    static let navigation = "navigation"
    static let deepLinkOpen = "deep_link_open"

    static let healthConnectSync = "health_connect_sync"
    static let healthDataRead = "health_data_read"
    static let healthDataWrite = "health_data_write"

    static let aiRequest = "ai_request"
    static let aiSuccess = "ai_success"
    static let aiFailed = "ai_failed"

    static let shareWorkout = "share_workout"
    static let shareProgress = "share_progress"
    static let shareAchievement = "share_achievement"
}

enum TelemetryParameters {
    static let screenName = "screen_name"
    static let action = "action"
    static let timestamp = "timestamp"
    static let duration = "duration"
    static let success = "success"

    static let userId = "user_id"
    static let userType = "user_type"

    static let workoutId = "workout_id"
    static let workoutName = "workout_name"
    static let workoutType = "workout_type"
    static let exerciseId = "exercise_id"
    static let exerciseName = "exercise_name"
    static let setNumber = "set_number"
    static let reps = "reps"
    static let weight = "weight"
    static let restTime = "rest_time"

    static let mealType = "meal_type"
    static let foodId = "food_id"
    static let foodName = "food_name"
    static let calories = "calories"
    static let protein = "protein"
    static let carbs = "carbs"
    static let fat = "fat"
    static let mealId = "meal_id"
    static let nutritionMethod = "nutrition_method"

    static let programId = "program_id"
    static let programName = "program_name"
    static let programType = "program_type"
    static let programDuration = "program_duration"
    static let programWeek = "program_week"
    static let programDay = "program_day"

    static let exerciseCategory = "exercise_category"
    static let exerciceMuscleGroup = "exercise_muscle_group"

    static let aiType = "ai_type"
    static let aiModel = "ai_model"
    static let aiInput = "ai_input"
    static let aiOutput = "ai_output"

    static let healthDataType = "health_data_type"
    static let healthValue = "health_value"
    static let healthUnit = "health_unit"

    static let errorMessage = "error_message"
    static let errorStackTrace = "error_stack_trace"
    static let errorCode = "error_code"

    static let navigationFrom = "navigation_from"
    static let navigationTo = "navigation_to"
    static let deepLinkPath = "deep_link_path"
}

File: android-native/app/src/main/java/com/example/kpkn/telemetry/TelemetryEvents.kt
package com.example.kpkn.telemetry

/**
 * Constants for Firebase Analytics events and parameters
 */
object TelemetryEvents {
    // App lifecycle events
    const val APP_OPEN = "app_open"
    const val APP_FOREGROUND = "app_foreground"
    const val APP_BACKGROUND = "app_background"
    
    // Authentication events
    const val LOGIN_SUCCESS = "login_success"
    const val LOGIN_FAILED = "login_failed"
    const val LOGOUT = "logout"
    
    // Training/Workout events
    const val WORKOUT_START = "workout_start"
    const val WORKOUT_END = "workout_end"
    const val WORKOUT_PAUSE = "workout_pause"
    const val WORKOUT_RESUME = "workout_resume"
    const val WORKOUT_CANCEL = "workout_cancel"
    const val SET_COMPLETE = "set_complete"
    const val REST_START = "rest_start"
    const val REST_END = "rest_end"
    const val EXERCISE_COMPLETE = "exercise_complete"
    
    // Nutrition events
    const val NUTRITION_OPEN = "nutrition_open"
    const val MEAL_LOG_START = "meal_log_start"
    const val MEAL_LOG_COMPLETE = "meal_log_complete"
    const val FOOD_SEARCH = "food_search"
    const val FOOD_ITEM_ADD = "food_item_add"
    const val NUTRITION_PARSE_PHOTO = "nutrition_parse_photo"
    const val NUTRITION_PARSE_SUCCESS = "nutrition_parse_success"
    const val NUTRITION_PARSE_FAILED = "nutrition_parse_failed"
    
    // Program events
    const val PROGRAM_SELECT = "program_select"
    const val PROGRAM_START = "program_start"
    const val PROGRAM_COMPLETE = "program_complete"
    const val PROGRAM_CANCEL = "program_cancel"
    
    // Exercise events
    const val EXERCICE_SEARCH = "exercise_search"
    const val EXERCICE_VIEW_DETAILS = "exercise_view_details"
    const val EXERCICE_ADD_TO_WORKOUT = "exercise_add_to_workout"
    
    // WikiLab events
    const val WIKILAB_OPEN = "wikilab_open"
    const val WIKILAB_EXERCISE_VIEW = "wikilab_exercise_view"
    
    // Settings events
    const val SETTINGS_OPEN = "settings_open"
    const val SETTINGS_CHANGE = "settings_change"
    const val THEME_CHANGE = "theme_change"
    const val LANGUAGE_CHANGE = "language_change"
    
    // Navigation events
    const val NAVIGATION = "navigation"
    const val DEEP_LINK_OPEN = "deep_link_open"
    
    // Health events
    const val HEALTH_CONNECT_SYNC = "health_connect_sync"
    const val HEALTH_DATA_READ = "health_data_read"
    const val HEALTH_DATA_WRITE = "health_data_write"
    
    // AI events
    const val AI_REQUEST = "ai_request"
    const val AI_SUCCESS = "ai_success"
    const val AI_FAILED = "ai_failed"
    
    // Sharing events
    const val SHARE_WORKOUT = "share_workout"
    const val SHARE_PROGRESS = "share_progress"
    const val SHARE_ACHIEVEMENT = "share_achievement"
}

object TelemetryParameters {
    // Common parameters
    const val SCREEN_NAME = "screen_name"
    const val ACTION = "action"
    const val TIMESTAMP = "timestamp"
    const val DURATION = "duration"
    const val SUCCESS = "success"
    
    // User parameters
    const val USER_ID = "user_id"
    const val USER_TYPE = "user_type"
    
    // Workout parameters
    const val WORKOUT_ID = "workout_id"
    const val WORKOUT_NAME = "workout_name"
    const val WORKOUT_TYPE = "workout_type"
    const val EXERCISE_ID = "exercise_id"
    const val EXERCISE_NAME = "exercise_name"
    const val SET_NUMBER = "set_number"
    const val REPS = "reps"
    const val WEIGHT = "weight"
    const val REST_TIME = "rest_time"
    
    // Nutrition parameters
    const val MEAL_TYPE = "meal_type" // breakfast, lunch, dinner, snack
    const val FOOD_ID = "food_id"
    const val FOOD_NAME = "food_name"
    const val CALORIES = "calories"
    const val PROTEIN = "protein"
    const val CARBS = "carbs"
    const val FAT = "fat"
    const val MEAL_ID = "meal_id"
    const val NUTRITION_METHOD = "nutrition_method" // manual, photo, barcode
    
    // Program parameters
    const val PROGRAM_ID = "program_id"
    const val PROGRAM_NAME = "program_name"
    const val PROGRAM_TYPE = "program_type"
    const val PROGRAM_DURATION = "program_duration"
    const val PROGRAM_WEEK = "program_week"
    const val PROGRAM_DAY = "program_day"
    
    // Exercise parameters
    const val EXERCISE_CATEGORY = "exercise_category"
    const val EXERCICE_MUSCLE_GROUP = "exercise_muscle_group"
    
    // AI parameters
    const val AI_TYPE = "ai_type" // nutrition, workout_advice, exercise_info
    /** Legacy event key kept for imported historical exports; Android emits local parser events. */
    const val AI_MODEL = "ai_model"
    const val AI_INPUT = "ai_input"
    const val AI_OUTPUT = "ai_output"
    
    // Health parameters
    const val HEALTH_DATA_TYPE = "health_data_type" // weight, fat, steps, exercise
    const val HEALTH_VALUE = "health_value"
    const val HEALTH_UNIT = "health_unit"
    
    // Error parameters
    const val ERROR_MESSAGE = "error_message"
    const val ERROR_STACK_TRACE = "error_stack_trace"
    const val ERROR_CODE = "error_code"
    
    // Navigation parameters
    const val NAVIGATION_FROM = "navigation_from"
    const val NAVIGATION_TO = "navigation_to"
    const val DEEP_LINK_PATH = "deep_link_path"
}
